go/database/mpt/io/parallel_visit_test.go

```diff
@@ -219,45 +219,68 @@ func TestVisit_Nodes_CannotOpenFiles(t *testing.T) {
 }
 
 func TestVisit_Nodes_CannotCloseSources(t *testing.T) {
-	injectedError := fmt.Errorf("injected error")
-
-	trie := createArchive(t)
-
-	nodeId, err := trie.GetBlockRoot(0)
-	if err != nil {
-		t.Fatalf("failed to get block root: %v", err)
-	}
-	if err := trie.Close(); err != nil {
-		t.Fatalf("failed to close archive: %v", err)
+	sourceFactories := []struct {
+		name       string
```

Comment:
nit: when your test cases have a `name` a nice trick is also to use a `map[string]struct{ ... }` and use the `name` as a key instead of a field. This gives a nice syntax for enumerating cases (`for name, test := range tests {`) and avoid name collisions (probably not that much of a problem).

Reply:
updated